pos_check_session_state/static/src/js/pos_check_session_state.js

```diff
@@ -0,0 +1,68 @@
+/******************************************************************************
+    Point Of Sale - Check Session State module for Odoo
+    Copyright (C) 2015-Today GRAP (http://www.grap.coop)
+    @author Sylvain LE GAL (https://twitter.com/legalsylvain)
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+openerp.pos_check_session_state = function (instance) {
+    module = instance.point_of_sale;
+
+    /* 
+        Define : New ErrorClosedSessionPopupWidget Widget.
+        This pop up will be shown if the current pos.session of the PoS is not
+        in an 'open' state;
+        The check will be down each 10 seconds;
+    */
+    module.ErrorClosedSessionPopupWidget = module.ErrorPopupWidget.extend({
+        template:'ErrorClosedSessionPopupWidget',
+
+        check_session_frequency: 10000,
+        session_name: '',
+
+        init: function(parent, options) {
+            var self = this;
+            this._super(parent,options);
+            this.intervalID = setInterval(function() {
+                var loaded = self.pos.fetch('pos.session', ['name','state'], [['id', '=', self.pos.get('pos_session').id]]) 
```

Comment:
This part raise an
![pos - fetch error](https://cloud.githubusercontent.com/assets/3407482/9441608/69101e56-4a77-11e5-9daf-2e96178ef36d.png)
 uggly recurring window if server is not available. (that is possible in PoS because it must works offline). 
Any idea to manage that case ? I tried with a try / catch without any success.

Kind regards.